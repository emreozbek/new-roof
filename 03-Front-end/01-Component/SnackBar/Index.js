import React, { Component } from "react";
import PropTypes from "prop-types";
import { Snackbar, IconButton } from "material-ui";
import { Close as CloseIcon } from "material-ui-icons";

class SnackbarComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: props.open
    };
  }
  componentWillReceiveProps(nextProps) {
    this.setState({ open: nextProps.open });
  }
  HandleClose() {
    this.setState({ open: false });
  }
  render() {
    const { message, vertical, horizontal, duration } = this.props;
    return (
      <Snackbar
        anchorOrigin={{
          vertical,
          horizontal
        }}
        open={this.state.open}
        autoHideDuration={duration}
        onClose={this.HandleClose.bind(this)}
        SnackbarContentProps={{
          "aria-describedby": "message-id"
        }}
        message={<span id="message-id">{message}</span>}
        action={[
          <IconButton
            key="close"
            aria-label="Close"
            color="inherit"
            onClick={this.handleClose.bind(this)}
          >
            <CloseIcon />
          </IconButton>
        ]}
      />
    );
  }
}
SnackbarComponent.propTypes = {
  message: PropTypes.string,
  vertical: PropTypes.string,
  horizontal: PropTypes.string,
  duration: PropTypes.number,
  open: PropTypes.bool.isRequired
};
SnackbarComponent.defaultProps = {
  message: "",
  vertical: "top",
  horizontal: "center",
  duration: 6000
};
export default SnackbarComponent;
