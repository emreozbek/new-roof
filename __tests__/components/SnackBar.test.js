import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SnackBar from "../../Client/Component/SnackBar/Index";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const snackBar = shallow(<SnackBar open={false} />);
  expect(snackBar).toMatchSnapshot();
});
