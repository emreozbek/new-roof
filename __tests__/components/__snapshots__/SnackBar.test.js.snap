// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withStyles(Snackbar)
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <withStyles(IconButton)
        aria-label="Close"
        color="inherit"
        onClick={[Function]}
      >
        <pure(Close) />
      </withStyles(IconButton)>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "top",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    >
      
    </span>
  }
  onClose={[Function]}
  open={false}
/>
`;
